src/perl.c

```diff
@@ -2390,6 +2384,11 @@ static void xs_init(pTHX) {
    * accessing any such variable (this is basically the same as using
    * tie() in Perl) */
   /* global strings */
+  struct {
```

Comment:
Why did this need to be moved?

Reply:
It was breaking the integration tests with the error message [1]:

src/perl.c:268:1: error: initializer element is not constant
 } g_strings[] = {{"Collectd::hostname_g", hostname_g}, {"", NULL}};
 ^

Since it was a global variable, it could not be initialized with hostname_g as type char*. Changing it to a local variable fixed the issue, and I didn't see anywhere else it was used.

On that note though, I suppose the "g_" prefix would no longer be appropriate. What do you think?

[1] https://ci.collectd.org/job/pull-requests-build-on-jessie-amd64-with-default-toolchain/1766/console